In [1]:
import pandas as pd
import numpy as np
from sklearn import svm
from sklearn.model_selection import train_test_split 
from sklearn.model_selection import cross_val_score
from sklearn import metrics
from sklearn.metrics import classification_report, confusion_matrix

# Reading the data

In [2]:
data = pd.read_csv('train_data.csv')
data.shape

(792, 17)

# Simple SVM Model

In [3]:
cols = ['PassengerId', 'Sex', 'Age', 'Fare', 'Pclass_1',
       'Pclass_2', 'Pclass_3', 'Family_size', 'Title_1', 'Title_2', 'Title_3',
       'Title_4', 'Emb_1', 'Emb_2', 'Emb_3']
y = data['Survived']
x = data[cols]
X_train, X_test, y_train, y_test = train_test_split(x, y, test_size=0.3, random_state=1)

clf = svm.SVC(kernel = 'linear')
# Fit the classifier to the data
clf.fit(X_train,y_train)

y_pred =clf.predict(X_test)
print("Accuracy:",metrics.accuracy_score(y_test, y_pred))

Accuracy: 0.7773109243697479


# SVM with kernel fucntion

In [4]:
svclassifier = svm.SVC(kernel='rbf', class_weight='balanced')
svclassifier.fit(X_train, y_train)

y_pred1 = svclassifier.predict(X_test)

print(confusion_matrix(y_test, y_pred))
print(classification_report(y_test, y_pred))

print("Accuracy:",metrics.accuracy_score(y_test, y_pred))


[[120  22]
 [ 31  65]]
              precision    recall  f1-score   support

           0       0.79      0.85      0.82       142
           1       0.75      0.68      0.71        96

    accuracy                           0.78       238
   macro avg       0.77      0.76      0.76       238
weighted avg       0.78      0.78      0.78       238

Accuracy: 0.7773109243697479


/opt/anaconda3/lib/python3.7/site-packages/sklearn/svm/base.py:193: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


#### I used the gaussian kernel function. I noticed that there was not much difference in the performance
    

#### Below are the accuracies for various models I have used:
#### 1. Logistic Regression : 82%
#### 2. Decision Tree : 78%
#### 3. K Nearest Neighbor: 57%
#### 4. Support Vector Machine: 

#### comapring the accuracies, we can say that for this datasest - Logistic regression performs the best.